python/ray/_private/runtime_env/validation.py

```diff
@@ -71,6 +73,10 @@ def __init__(self,
         # contain all supported keys; values will be set to None if
         # unspecified. However, if all values are None this is set to {}.
         self._dict = dict()
+        self.known_fields: Set[str] = {
+            "working_dir", "conda", "pip", "uris", "containers", "env_vars",
+            "_ray_release", "_ray_commit", "_inject_current_ray", "plugins"
+        }
```

Comment:
does this need to be a field of `self`? prefer to just make it a global constant